### ISSUE TITLE
about concept quantization

### ISSUE THREAD

**Issue by qizhust** [vacancy/NSCL-PyTorch-Release#15]
Hi, in your paper, Pr[object i is Red] is given by shifted and scaled sigmoid function, but there seems no sigmoid function in your code as shown below

https://github.com/vacancy/NSCL-PyTorch-Release/blob/ef493d58a986dcb1ea16a23183a57119abeaff34/nscl/nn/reasoning_v1/concept_embedding.py#L181

If so, the range of the 'logits' variable can be a problem when it adds with the 'belong' vector. Could you explain more about this? Thx!

**Comment by vacancy** [comment]
Thanks for bringing up this. Yes, the implementation here does not exactly follow the equation in the paper. 
To correct that, one should add an additional `logsigmoid` operation before it got added up with the `belong` vector.
And we should also replace the BCEWithLogit loss with a `BCE(exp(log_logits), label)` loss.

However, in practice, I feel that the current implementation won't significantly hurt the training.
To understand that, interpret the `logsumexp` as a "soft max" operation. So the additional `belong` vector with adjust the logits so that if concept c does not belong to attribute a, it will has a logit value less than 0. The logsumexp will operation will pick the largest possible logit.

**Comment by qizhust** [comment]
Thanks a lot! I agree with you that the effect can be the same without a sigmoid function.
